components/net/filemanager_thread.rs

```diff
@@ -171,6 +176,176 @@ impl FileManager {
             },
         }
     }
+
+    pub fn fetch_file_in_chunks(
+        &self,
+        done_sender: Sender<Data>,
+        mut reader: BufReader<File>,
+        res_body: ServoArc<Mutex<ResponseBody>>,
+        cancellation_listener: Arc<Mutex<CancellationListener>>,
+        range: RelativePos,
+    ) {
+        self.thread_pool.upgrade().and_then(|pool| {
+            pool.spawn(move || {
+                loop {
+                    if cancellation_listener.lock().unwrap().cancelled() {
+                        *res_body.lock().unwrap() = ResponseBody::Done(vec![]);
+                        let _ = done_sender.send(Data::Cancelled);
+                        return;
+                    }
+                    let length = {
+                        let buffer = reader.fill_buf().unwrap().to_vec();
+                        let mut buffer_len = buffer.len();
+                        if let ResponseBody::Receiving(ref mut body) = *res_body.lock().unwrap() {
+                            let offset = usize::min(
+                                {
+                                    if let Some(end) = range.end {
+                                        // HTTP Range requests are specified with closed ranges,
```

Comment:
you can use inclusive rust ranges here if we want to

Reply:
I've filed an issue at https://github.com/servo/servo/issues/25751 because since this code is only moved around in this PR, I would rather not change it here. I've marked the issue "easy", perhaps you can elaborate further there?

Reply:
Works.